riscv-runtime/src/std_support.rs

```diff
@@ -0,0 +1,79 @@
+// The following functions are expected by `std` and `panic_abort` to exist for
+// the zkvm target, and we must provide them. The list was taken from file
+// "src/sys/pal/zkvm/abi.rs" in the `std` source code, filtered for functions
+// that are actually used.
+
+// We don't provide the sys_alloc_aligned(), despite it being used as a default
+// allocator, because we already define a global allocator in the `allocator`
+// module. It is used in both `std` and `no_std` modes.
+
+use core::{alloc::Layout, arch::asm, slice};
+
+use powdr_riscv_syscalls::Syscall;
+
+use crate::io::write_slice;
+
+#[no_mangle]
+extern "C" fn sys_rand(buf: *mut u32, words: usize) {
+    let slice = unsafe { slice::from_raw_parts_mut(buf, words) };
+
+    // I promise I rolled a fair die and this is the result.
```

Comment:
maybe replace this line with an explanation, similar to what you wrote in the chat?

Reply:
Done.